parlai/agents/transformer/modules.py

```diff
@@ -42,6 +42,7 @@ def _build_encoder(opt, dictionary, embedding=None, padding_idx=None, reduction=
         learn_positional_embeddings=opt.get('learn_positional_embeddings', False),
         embeddings_scale=opt['embeddings_scale'],
         reduction=reduction,
+        n_positions=opt.get('truncate', 1024)
```

Comment:
Hmm - will this fail if token vectors are not truncated and one vector has length > 1024?

Reply:
Also - should we un-hardcode this in the decoder (line 324) as well?

Reply:
Token vectors should be truncated if you specify `truncate`. Will update so it's not hardcoded in the decoder either